# Ct_pop_gen_project

This repository contains scripts and files used to perform the analyses and make the figures and tables associated with the *Caenorhabditis tropicalis* population genetics manuscript 

- The "data" folder contains the raw datasets from public database (*e.g.* CaeNDR)
- The "plots" folder contains all the figures generated by the scripts
- The "processed_data" folder contains all the intermediate results files generated during the analysis
- The "scripts" folder contains all the codes used for generating figures and tables
- The "tables" folder contains all the tables generated by the scripts

<br>
<br>

---

<br>

## How to use the scripts in this repository to plot figures and generate tables in the manuscript
This section shows the names and order of the scripts used to plot figures and generate tables. <br>
Notes: The "&#8594;" reprensents the order of scripts. The “+” means you should use them both/all in that step. 

<br>

### Main Figures & Tables

#### Figure 1. Global distribution of 518 *C. tropicalis* isotype reference strains
- Figure 1 a-f: Plot_isotype_tropicalis_global_map.R 
- Figure 1 g-i: Plot_PCA_vcf.R

#### Figure 2. The ML tree of 518 *C. tropicalis* isotype reference strains generated from variants pruned with LD r<sup>2</sup> values less than or equal to 0.9
- Figure 2: Calculate_C_tro_vcf_to_pyh.sh + vcf2phylip.py &#8594; <br>
  Calculate_C_tro_LD_pruned_0.9_pyh_to_tree.sh &#8594; <br>
  Plot_tree_in_lat_with_label.R <br> 
  Notes: The vcf2phylip.py can be found in https://github.com/edgardomortiz/vcf2phylip

#### Figure 3. Diversity statistic calculated across global *C. tropicalis* isotypes
- Figure 3: Plot_tro_pi_theta_d.R

#### Figure 4. LD decay for all *C. tropicalis* isotype reference strains across all autosomes
- Figure 4: Calculate_LD_per_Mb_all.sh + Calculate_LD_per_Mb_geo.sh &#8594; <br>
  Plot_LD_geo_all.R

#### Figure 5. Hyper-variable regions (HVRs) in *C. tropicalis*
- Figure 5 b, c: Calculate_HVRs_TajimasD_bed_file.R &#8594; <br>
  Calculate_vcf_to_zarr_HVRs.sh &#8594; <br>
  Calculate_zarr_to_pi_theta_d_HVRs.sh &#8594; <br>
  Plot_HVRs_TajimasD.R

#### Figure 6. Relatedness and frequency of 518 *C. tropicalis* isotype reference strains at the Medea regions
- Figure 6: Calculate_C_tro_Medea_vcf.sh &#8594; <br>
  Plot_Medea_vcf2tree.R

#### Figure 7. Human Impact Index of nematode sampling sites
- Figure 7: Plot_human_footprint_index.R

#### Table 1. Effective population size (N<sub>e</sub>) for *Caenorhabditis* spp. samples from different studies
- Table 1: <br>
  *C. tropicalis*:
  Calculate_vcf_to_zarr_geo.sh &#8594; <br>
  Calculate_zarr_to_pi_theta_d_geo.sh <br>
  *C. elegans*:
  Calculate_Ce_vcf_to_zarr.sh  &#8594; <br>
  Calculate_Ce_zarr_to_pi_theta_d.sh <br>
  *C. briggsae*:
  Calculate_Cb_zarr_to_pi_theta_d.sh <br>
  Calculate_Cb_vcf_to_zarr.sh <br>
  Then, generate the table with all the output above: Table_isotypes_Ne_Outcrossing rate.R

#### Table 2. Outcrossing rate of isotype reference strains from different sampling sites
- Table 2: Calculate_vcf_to_zarr_geo.sh &#8594; <br>
  Calculate_zarr_to_pi_theta_d_geo.sh + Calculate_LD_per_Mb_all.sh + Calculate_LD_per_Mb_geo.sh  &#8594; <br>
  Table_isotypes_Ne_Outcrossing rate.R


<br>
<br>

### Supplemental Figures & Tables

#### Figure S1. Global distribution of 690 *C. tropicalis* strains
- Figure S1: Plot_strains_tropicalis_global_map.R

#### Figure S2. Geographical distance between strains within an isotype
- Figure S2 a: Plot_Vincenty_geo_Ce.R
- Figure S2 b: Plot_Vincenty_geo_Ct.R

#### Figure S3-S5. The ML trees of 518 *C. tropicalis* isotype reference strains generated from LD-pruned variants with r<sup>2</sup> values less than or equal to 0.7-0.9
- Figure S3-S5: Calculate_C_tro_vcf_to_pyh.sh + vcf2phylip.py &#8594; <br>
  Calculate_C_tro_LD_pruned_0.6_pyh_to_tree.sh + Calculate_C_tro_LD_pruned_0.7_pyh_to_tree.sh + Calculate_C_tro_LD_pruned_0.8_pyh_to_tree.sh &#8594; <br>
  Plot_tree_in_lat_with_label.R <br>
  Notes: The vcf2phylip.py can be found in https://github.com/edgardomortiz/vcf2phylip

#### Figure S6. Unrooted maximum likelihood trees of 518 *C. tropicalis* isotype reference strains generated from LD-pruned variants
- Figure S6: Plot_equal angle_unrooted_tree.R

#### Figure S7. Scatter plot shows significant positive correlation between geographic distance and phylogenetic distance
- Figure S7: Plot_phy_geo.R

#### Figure S8. Genetic distance (*D<sub>xy</sub>*) correlation with geographic distance and regional variation
- Figure S8: Calculate_C_tro_dxy(10kb)_only_among_geo.sh  &#8594; <br>
  Plot_dxy_geo_change_in_dxy_per_km.R

#### Figure S9. Pairwise fixation index (*F<sub>st</sub>*) among *C. tropicalis* from different geographic regions
- Figure S9: Calculate_C_tro_fst_geo.sh  &#8594; <br>
  Plot_fst_geo.R

#### Figure S10. LD decay for all autosomes
- Figure S10: Calculate_LD_per_Mb_all.sh + Calculate_LD_per_Mb_geo.sh  &#8594; <br>
  Plot_LD_geo_all_chromosome.R

#### Figure S11. Species wide (*D<sub>xy</sub>*)  of *C. tropicalis* 
- Figure S11: Calculate_C_tro_dxy_10kb_chunk_I_to_X.sh + Calculate_C_tro_dxy_10kb_chunk_V_1.sh + Calculate_C_tro_dxy_10kb_chunk_V_2.sh  &#8594; <br>
  Calculate_C_tro_dxy_10kb_stat.sh &#8594; <br>
  Plot_dxy_species_wide.R

#### Figure S12-S16. Relatedness of 518 *C. tropicalis* isotype reference strains at five different maternal effect regions
- Figure S12-S16: Plot_TAs_vcf2tree.R

#### Table S4 Diversity statistic (*π*, *θ*, Tajima's D) for all isotype reference strains and for each geographic region
- Table S4: Calculate_vcf_to_zarr_geo.sh &#8594; <br>
  Calculate_zarr_to_pi_theta_d_geo.sh &#8594; <br>
  Table_geo_p_theta.R

#### Table S5 Effective population size (N<sub>e</sub>) for all isotype reference strains and for each geographic region except for the two under-sampled regions, Africa and Australia.
- Table S5: Calculate_vcf_to_zarr_geo.sh &#8594; <br>
  Calculate_zarr_to_pi_theta_d_geo.sh + Calculate_LD_per_Mb_all.sh + Calculate_LD_per_Mb_geo.sh  &#8594; <br>
  Table_isotypes_Ne_Outcrossing rate.R



