# Input data was kindly generated by Ryan

rm(list = ls())


library(dplyr)
library(ggplot2)


library(RColorBrewer)
display.brewer.all()
display.brewer.pal(8,"Set3")
brewer.pal(8,"Set3")[4:6]
display.brewer.pal(9,"Greys")
brewer.pal(9,"Greys")

species_wide_diversity_raw<-read.csv("../processed_data/pi_theta_d/sub_region_diversity.csv",
         header = TRUE,
         row.names = 1)

species_wide_diversity<- species_wide_diversity_raw %>% 
  dplyr::mutate(sub_region = ifelse(sub_region == "left_arm", "left arm", sub_region)) %>% 
  dplyr::mutate(sub_region = ifelse(sub_region == "left_tip", "left tip", sub_region)) %>% 
  dplyr::mutate(sub_region = ifelse(sub_region == "right_arm", "right arm", sub_region)) %>% 
  dplyr::mutate(sub_region = ifelse(sub_region == "right_tip", "right tip", sub_region))
  
species_wide_diversity$sub_region <- factor(species_wide_diversity$sub_region, 
                                            levels = c("left tip", "left arm", "center", "right arm", "right tip", "full"))



colnames(species_wide_diversity)

# pi 
plot_pi<-species_wide_diversity %>%
  dplyr::select(chrom,sub_region,pi) 

# theta
plot_theta<-species_wide_diversity %>%
  dplyr::select(chrom,sub_region,theta)
# d
plot_d<-species_wide_diversity %>%
  dplyr::select(chrom,sub_region,d)




# plot pi heatmap
p_pi <- ggplot(plot_pi, aes(x = sub_region, y = chrom, fill = pi)) +
  ggplot2::geom_tile(color = "white") +
  ggplot2::scale_fill_gradient(low = "lightyellow", high = "orange") +  
  ggplot2::theme_minimal() +  
  # ggplot2::labs(x = "Region", y = "Population") +
  ggplot2::theme(panel.grid.major = element_blank(),
                 panel.grid.minor = element_blank())+
  # theme(legend.position = c(0.2, 0.7))+
  theme(legend.position = "bottom",
        legend.title = element_blank(),
        legend.margin = margin(t = -0.1, unit = "cm"),
        legend.key.size = unit(0.3, "cm"),
        legend.text = element_text(size = 6, angle = 45, hjust = 1, vjust = 1.4))+
  ggplot2::ggtitle(expression("Nucleotide diversity (" * pi * ")"))+
  ggplot2::theme(plot.title = element_text(hjust = 0.5, size = 8, vjust = -2, color = "black"),
                 axis.text.x = element_text(size = 6, color = "black"),  
                 axis.text.y = element_text(size = 6, color = "black"))+
  # ggplot2::ggtitle("Nucleotide diversity (\u03C0)")+
  ggplot2::labs(x = NULL, y = NULL)+
  ggplot2::geom_text(aes(label = round(pi, 5)), color = "black", size = 1.9)  # add text labels

p_pi






# plot theta heatmap
p_theta <- ggplot(plot_theta, aes(x = sub_region, y = chrom, fill = theta)) +
  ggplot2::geom_tile(color = "white") +
  ggplot2::scale_fill_gradient(low = "lightyellow", high = "orange") +  
  ggplot2::theme_minimal() +  
  # ggplot2::labs(x = "Region", y = "Population") +
  ggplot2::theme(panel.grid.major = element_blank(),
                 panel.grid.minor = element_blank())+
  # theme(legend.position = c(0.2, 0.7))+
  theme(legend.position = "bottom",
        legend.title = element_blank(),
        legend.margin = margin(t = -0.1, unit = "cm"),
        legend.key.size = unit(0.3, "cm"),
        legend.text = element_text(size = 6, angle = 45, hjust = 1, vjust = 1.4))+
  ggplot2::ggtitle(expression("Watterson's " * theta * ""))+
  ggplot2::theme(plot.title = element_text(hjust = 0.5, size = 8, vjust = -2, color = "black"),
                 axis.text.x = element_text(size = 6, color = "black"),  
                 axis.text.y = element_text(size = 6, color = "black"))+
  # ggplot2::ggtitle("Nucleotide diversity (\u03C0)")+
  ggplot2::labs(x = NULL, y = NULL)+
  ggplot2::geom_text(aes(label = round(theta, 5)), color = "black", size = 1.9)  # add text labels

p_theta







# plot d heatmap
p_d <- ggplot(plot_d, aes(x = sub_region, y = chrom, fill = d)) +
  ggplot2::geom_tile(color = "white") +
  ggplot2::scale_fill_gradient(low = "lightyellow", high = "orange") +  
  ggplot2::theme_minimal() +  
  # ggplot2::labs(x = "Region", y = "Population") +
  ggplot2::theme(panel.grid.major = element_blank(),
                 panel.grid.minor = element_blank())+
  # theme(legend.position = c(0.2, 0.7))+
  theme(legend.position = "bottom",
        legend.title = element_blank(),
        legend.margin = margin(t = -0.1, unit = "cm"),
        legend.key.size = unit(0.3, "cm"),
        legend.text = element_text(size = 6, angle = 45, hjust = 1, vjust = 1.4))+
  ggplot2::ggtitle(expression("Tajima's D"))+
  ggplot2::theme(plot.title = element_text(hjust = 0.5, size = 8, vjust = -2, color = "black"),
                 axis.text.x = element_text(size = 6, color = "black"),  
                 axis.text.y = element_text(size = 6, color = "black"))+
  # ggplot2::ggtitle("Nucleotide diversity (\u03C0)")+
  ggplot2::labs(x = NULL, y = NULL)+
  ggplot2::geom_text(aes(label = round(d, 3)), color = "black", size = 1.9)  # add text labels

p_d


# plot a-c
library(ggpubr)
p_a_c <- ggpubr::ggarrange(p_pi, p_theta, p_d, 
                           ncol = 3,       
                           nrow = 1,        
                           labels = c("a", "b", "c"))     
p_a_c

# ggsave("p_a_c.pdf", plot = p_a_c, width = 7.5, height = 2.5, units = "in")







###################################
######### genome wide loess #######
###################################


# Input data was kindly generated by Ryan


# install.packages("extrafont")
# font_import("Arial")

# rm(list = ls())

library(extrafont)
library(dplyr)
library(ggplot2)

library(RColorBrewer)
display.brewer.all()
display.brewer.pal(8,"Set3")
brewer.pal(8,"Set3")[4:6]
display.brewer.pal(9,"Greys")
brewer.pal(9,"Greys")

window_diversity<-read.csv("../processed_data/pi_theta_d/chromosome_windows_diversity.csv",
                           header = TRUE,
                           row.names = 1)

#Plot all windowed diversity stats
windowed_div_stats <- function(windows_df){
  diversity <- windows_df %>%
    dplyr::filter(stat_type == "pi" | stat_type == "theta" | stat_type == "d") %>%
    dplyr::mutate(stat_type = if_else(stat_type == "d", "Tajima's D", stat_type)) %>%
    dplyr::mutate(stat_type = if_else(stat_type == "pi", "Nucleotide diversity (\u03C0)", stat_type)) %>%
    dplyr::mutate(stat_type = if_else(stat_type == "theta", "Watterson's \u03B8", stat_type)) %>%
    # normalize the Reference genome position
    group_by(chrom) %>%
    mutate(x_normalized = (x - min(x)) / (max(x) - min(x)))
  
  diversity$stat_type <- factor(diversity$stat_type, levels = c("Nucleotide diversity (\u03C0)", "Watterson's \u03B8", "Tajima's D", "mis"))
  
  
  
  
  
  p1 <- ggplot2::ggplot(data = diversity, mapping = aes(x = x_normalized, y = stat,color= stat_type))+
    geom_point(color = "lightgray", size = 0.02, alpha = 0.8)+
    facet_grid(stat_type ~ chrom, scales = 'free')+
    xlab("Normalized genome position")+
    ylab("Diversity statistic")+
    geom_smooth(method = "loess", se = FALSE,alpha = 0.8,color = "orange") +
    # scale_color_manual(values = c("Tajima's D" = "#FB8072",
    #                               "Nucleotide diversity (\u03C0)" = "#FDB462",
    #                               "Watterson's \u03B8" = "#80B1D3")) +
    
    theme_bw()+
    theme(legend.position = "none")+
    theme(panel.grid.major = element_blank(),
          panel.grid.minor = element_blank())+
    theme(strip.background = element_blank())+
    scale_x_continuous(breaks = c(0, 0.5, 1))+
    theme(axis.title.y = element_text(face = "bold", size = 9))+
    theme(axis.title.x = element_text(face = "bold", size = 9))+
    theme(strip.text = element_text(face = "bold", size = 7,color = "#525252"))
  
  
  
  plots <- p1
  return(plots)
}

result_plots <- windowed_div_stats(window_diversity)
result_plots

# save image 4.5 * 7.5 inches
# ggsave("plot.pdf", plot = result_plots, width = 7.5, height = 4.5, units = "in")








### merge all
library(ggpubr)

plot_all<-ggarrange(
  ggpubr::ggarrange(p_pi, p_theta, p_d, nrow = 1, labels = c("a", "b", "c")),
  result_plots,
  heights = c(0.357, 0.643),  
  labels = c("","d"),  
  nrow = 2  
)
plot_all

ggsave("../plots/Plot_pi_theta_d.pdf", plot = plot_all, width = 7.5, height = 7, units = "in")








