# the HVRs diversity file was genrated from the zarr_to_pi_theta_d_HVRs.sh and vcf_to_zarr_HVRs.sh
# HVRs bed file was kindly generated by Nic and further processed with HVRs_TajimasD_bed_file.R script

rm(list = ls())

library(dplyr)
library(ggplot2)
library(ggalluvial)
# library(purrr)


HVRs_div_raw<-read.csv("../processed_data/pi_theta_d/HVRs_pi_theta_d/chromosome_windows_diversity.csv",
                          header = TRUE)


HVRs_TajimasD <- HVRs_div_raw %>%
  dplyr::filter(stat_type == "d") %>% 
  na.omit()
  

#Plot all windowed diversity stats

HVRs_TajimasD_plot <- HVRs_TajimasD %>%
  dplyr::mutate(stat_type = if_else(stat_type == "d", "Tajima's D", stat_type)) %>%
  dplyr::mutate(div_stat = if_else(div_stat == "True", "HVRs", "Non-HVRs")) %>%
  # classify D types
  mutate(stat_category = case_when(
    stat <= 0 ~ "<=0",
    stat > 0 & stat <= 1 ~ "0-1",
    stat > 1 & stat <= 2 ~ "1-2",
    stat > 2 & stat <= 3 ~ "2-3",
    stat > 3 ~ ">3")) %>% 
  mutate(stat_category = factor(stat_category, levels = c("<=0", "0-1", "1-2", "2-3", ">3"))) %>% 
  # normalize the Reference genome position
  dplyr::group_by(chrom) %>%
  dplyr::mutate(x_normalized = (x - min(x)) / (max(x) - min(x)))

  
HVRs.colors <- c("#DD3913", "darkgrey")
p1 <- ggplot2::ggplot(data = HVRs_TajimasD_plot, mapping = aes(x = x_normalized, y = stat, fill = div_stat))+
  geom_point(shape=21, size = 0.7, alpha = 0.8, stroke = 0)+
  scale_fill_manual(name = "Group", values = HVRs.colors, labels = c("HVRs", "Non-HVRs")) + 
  facet_grid(stat_type ~ chrom, scales = 'free')+
  xlab("Normalized genome position")+
  ylab("Tajima's D")+
  # geom_smooth(method = "loess", se = FALSE,alpha = 0.8,color = "#BEBADA") +
  # scale_color_manual(values = c("Tajima's D" = "#FB8072",
  #                               "Nucleotide diversity (\u03C0)" = "#FDB462",
  #                               "Watterson's \u03B8" = "#80B1D3")) +
  
  theme_bw()+
  theme(legend.position = "none")+
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank())+
  theme(strip.background = element_blank())+
  scale_x_continuous(breaks = c(0, 0.5, 1))+
  theme(axis.title.y = element_text(face = "bold", size = 9))+
  theme(axis.title.x = element_text(face = "bold", size = 9))+
  theme(strip.text = element_text(face = "bold", size = 7,color = "#525252"))
p1




# Plot stacked bar for each categray 
# generate a table to plot figure 
summary_table <- HVRs_TajimasD_plot %>%
  group_by(stat_category, div_stat) %>%
  summarise(count = n(), .groups = 'drop') %>%
  group_by(stat_category) %>%
  mutate(percentage = count / sum(count) * 100)




# plot
p2<-ggplot2::ggplot(data = summary_table, aes(x = stat_category, y = percentage, alluvium = div_stat, stratum = div_stat)) +
  geom_alluvium(aes(fill = div_stat), alpha = 0.4, width = 0.6) + 
  geom_stratum(aes(fill = div_stat), width = 0.7, size = 0.2) +
  ggplot2::labs(x = "Tajima's D",
                y = "Percentage (%)",
                # title = "Percentage of HVRs and Non-HVRs",
                fill = "Type") +
  scale_fill_manual(values = HVRs.colors) +
  theme_bw()+
  theme(legend.position = "right")+
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank())+
  theme(strip.background = element_blank())+
  theme(axis.title.y = element_text(face = "bold", size = 9))+
  theme(axis.title.x = element_text(face = "bold", size = 9))+
  theme(strip.text = element_text(face = "bold", size = 7,color = "#525252"))+
  geom_text(aes(label = sprintf("%.1f%%", percentage)), 
            position = position_stack(vjust = 0.5), 
            size = 3)+
  coord_flip()
p2




# Arrange p1 and p2 side by side with specified proportions
library(ggpubr)


# combined_plot <- ggarrange(p1, p2, 
#                            ncol = 2, 
#                            widths = c(0.7, 0.3)) # 70% for p1 and 30% for p2

combined_plot <- ggpubr::ggarrange(
  p1, p2, 
  ncol = 1, 
  heights = c(0.6, 0.4), 
  labels = c("b", "c")
)
combined_plot

ggsave("../plots/HVRs_TajimasD.pdf", plot = combined_plot, width = 7.5, height = 4, units = "in")



# 
# # calculate mean value of Tajimas'D within HVRs
# HVRs_TajimasD_mean <- HVRs_div_raw %>%
#   filter(stat_type == "d" & div_stat == "True") %>% 
#   na.omit() %>% 
#   select(stat) %>% 
#   pull() %>% 
#   mean()
# 
# HVRs_TajimasD_mean
# # [1] 0.1446845
# 
# 

